.github/workflows/gh-pages.yml

```diff
@@ -0,0 +1,22 @@
+name: docs deploy
+
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v3
+        with:
+          python-version: 3.10.4
+      - run: |
```

Comment:
removed from Makefile, I am not sure if it is working, site doesnt seem to be updated.
the workflow succeeded though.

Reply:
Also note that when this get merged we should prob prefer to run this workflow on release only, i wasn't doing this for my ease.